net/tstun/wrap.go

```diff
@@ -543,7 +544,7 @@ func (t *Wrapper) Read(buffs [][]byte, sizes []int, offset int) (int, error) {
 	if !ok {
 		return 0, io.EOF
 	}
-	if res.err != nil {
+	if res.err != nil && len(res.data) == 0 {
```

Comment:
the next line has `res.data == nil` ... does len 0 vs nil mean different things?

Reply:
clarified in the docs on `type tunVectorReadResult`